src/fidesops/task/refine_target_path.py

```diff
@@ -48,33 +65,40 @@ def build_incoming_refined_target_paths(
 
 
 def refine_target_path(
-    row: Dict[str, Any], target_path: List[str], only: List[Any]
+    row: Row, target_path: List[str], only: Optional[List[Any]] = None
 ) -> DetailedPath:  # Can also return a list of DetailedPaths if there are multiple matches.
     """
     Recursively modify the target_path to be more detailed path(s) to the referenced data. Instead of just strings,
-    the path will be expanded to include indices where applicable.
+    the path will expand to include indices where applicable.
 
     :param row: Record retrieved from a dataset
     :param target_path: A list of strings representing the path to the desired field.
-    :param only: A list of values that were used to build the query.
-    :return: A list or a list of lists containing more detailed path(s) to the data in "only".  If there
-    was just one path, we return one list.
+    :param only: A list of values that you want to match. If you want to match all possible values, you can omit "only".
+    :return: A list or a list of lists containing more detailed path(s) to the data in "only". If data isn't limited in
+     "only", we expand the target_path to all possible target_paths.  If there was just one path, we return one list.
 
     :Example:
-    In this example, path ["A", "B", "C"] points to two elements that match values "F" or "G".  We update
-    the path to insert the indices to locate the appropriate value.
+    In this example, we want to expand path ["A", "B", "C"] to paths that match values "F" or "G".  We update
+    the path to insert the indices to target the specific element.
 
     refine_target_path({"A": {"B": [{"C": "D"}, {"C": "F"}, {"C": "G"}]}}, ["A", "B", "C"], only=["F", "G"])
 
     [["A", "B", 1, "C"], ["A", "B", 2, "C"]]
+
+    ------------------------------------------------------------------------------------------------------------------
+    :Example:
+    Here, we instead want to expand the paths to all possible paths, so we don't specify "only".
+    refine_target_path({"A": {"B": [{"C": "D"}, {"C": "F"}, {"C": "G"}]}}, ["A", "B", "C"])
+
+    [['A', 'B', 0, 'C'], ['A', 'B', 1, 'C'], ['A', 'B', 2, 'C']]
     """
     try:
         current_level = target_path[0]
         current_elem = row[current_level]
-    except KeyError:  # FieldPath not found in record, this is expected to happen when data doesn't exist in collection
-        return []
-    except (IndexError, TypeError):  # No field path or invalid field path
-        logger.warning(f"Error with locating {target_path} on row")
+    except (IndexError, TypeError, KeyError):
+        # KeyError - FieldPath not found, expected when data doesn't exist in collection
+        # IndexError - No FieldPath
+        # TypeError - Bad FieldPath, expected when we're trying to select a path we've already eliminated
         return []
```

Comment:
It would be good to continue warning here I think. Another option could be to use a different class of object (that can still exist in our Redis cache) for placeholders, then the delineation would be clearer since they wouldn't both be strings

Reply:
OK, restored the warning, again this warning is expected if we are masking an array where some of the elements have been marked with placeholder text, so they will be false positives.  We can tweak later if we are seeing this too much.